pkg/deploy/deploy.go

```diff
@@ -242,19 +242,34 @@ func (d *deployer) removeOldMetricAlerts(ctx context.Context) error {
 // from d.config.Configuration.
 func (d *deployer) getParameters(ps map[string]interface{}) *arm.Parameters {
 	m := map[string]interface{}{}
-
 	v := reflect.ValueOf(*d.config.Configuration)
 	for i := 0; i < v.NumField(); i++ {
-		m[strings.SplitN(v.Type().Field(i).Tag.Get("json"), ",", 2)[0]] = v.Field(i).Interface()
+		if v.Field(i).IsNil() {
+			continue
+		}
+
+		switch value := v.Field(i).Interface().(type) {
+		case *string:
+			m[strings.SplitN(v.Type().Field(i).Tag.Get("json"), ",", 2)[0]] = *value
+		default:
+			m[strings.SplitN(v.Type().Field(i).Tag.Get("json"), ",", 2)[0]] = value
+		}
```

Comment:
Maybe I'm wrong, but this might not be necessary. This should be enough:

```go
if v.Field(i).IsNil() {
	continue
}

m[strings.SplitN(v.Type().Field(i).Tag.Get("json"), ",", 2)[0]] = v.Field(i).Interface()
```

Later we convert this map into `arm.Parameters`. And it will jsut be marshalled into json. `encoding/json` should handle both string and pointer to a string. The important bit, as far as I understand, is not to add `nil`s  into parameters.

Reply:
Also what's implemented currently is brittle as soon as someone as someone introduces an int field, so +1 on going with what m1kola is suggesting.